### PR TITLE
ICT-107-video-upload-error

### PR DESCRIPTION
**Pull request by randyhoulahan**
Upload error and other ux improvements.

### CODE REVIEW

File: app/directives/kronos/participant.js [hunk 1]
```diff
@@ -267,6 +271,7 @@ app.directive('participant', ['$http','$timeout','conferenceService','$filter','
           $scope.onUpload=onUpload
 
           function onUpload (params, file, error){
+            if(error) return
```
Comment (rqk2k) [inline app/directives/kronos/participant.js:274]: Just wondering if this is the proper error handling.

**Review by rqk2k — APPROVED** [review]

File: app/directives/file.js [hunk 2]
```diff
@@ -49,26 +49,29 @@ import sharedT from '~/i18n/shared/index.js';
...
+                                $scope.hasError=translateError(err);
+
                                 files = files.concat([{ error: err }]);
 
                                 setViewValue(files);
 
                                 onUpload(htmlFile, null, err);
 
-                                res.data;
+                                $scope.loading = false;
```
Comment (blaisefonseca) [inline app/directives/file.js:73]: move this to finally

File: app/directives/kronos/participant.js [hunk 3]
```diff
@@ -179,11 +181,13 @@ app.directive('participant', ['$http','$timeout','conferenceService','$filter','
 
         function save(){
           validateRequireUploads()
+
           if($scope.editForm.$invalid) {
-            $scope.editForm.$submitted=true
-            $scope.error = { status: 4000 }
+            $scope.editForm.$submitted=true;
+            $scope.error = { status: 4000 };
```
Comment (blaisefonseca) [inline app/directives/kronos/participant.js:187]: should be `400` for badRequest

File: app/directives/file.js (hunk 2 above)
Comment (randyhoulahan) [reply app/directives/file.js]: done

File: app/directives/kronos/participant.js (hunk 3 above)
Comment (randyhoulahan) [reply app/directives/kronos/participant.js]: changed

**Review by sbilodeau — APPROVED** [review]

**Pull request merged**